portal/server/helpers/pageSpecificValidationErrors.js

```diff
@@ -4,126 +4,115 @@
 // - - the entities 'Check your answers' page (AKA 'Preview' page) has been viewed
 // - - one of entities forms has had a value submitted
 // by having pageSpecificValidationErrors logic in the UI, we remain decoupled = require(the API).
-
+const cloneDeep = require('lodash/cloneDeep')
 const errorHref = require('./errorHref');
 const generateErrorSummary = require('./generateErrorSummary');
-const {
-  requiredFieldsArray,
-  filterErrorList,
-} = require('./pageFields');
+const { requiredFieldsArray, filterErrorList } = require('./pageFields');
 
 const allFieldsArray = (fields) => {
-  const { OPTIONAL_FIELDS } = fields;
-  const allFields = requiredFieldsArray(fields);
+ const { OPTIONAL_FIELDS } = fields;
+ const allFields = requiredFieldsArray(fields);
 
-  if (OPTIONAL_FIELDS) {
-    allFields.push(...OPTIONAL_FIELDS);
-  }
+ if (OPTIONAL_FIELDS) {
+   allFields.push(...OPTIONAL_FIELDS);
+ }
 
-  return allFields;
+ return allFields;
 };
 
 const shouldReturnRequiredValidation = (fields, fieldValues) => {
-  const allFields = allFieldsArray(fields);
-
-  if (fieldValues.viewedPreviewPage) {
-    return true;
-  }
-
-  const totalFieldValues = Object.keys(fieldValues).filter((fieldName) =>
-    allFields.includes(fieldName)
-    && fieldValues[fieldName] !== null
-    && fieldValues[fieldName].length > 0);
-
-  if (totalFieldValues.length > 0) {
-    return true;
-  }
-
-  return false;
+ const allFields = allFieldsArray(fields);
+
+ if (fieldValues.viewedPreviewPage) {
+   return true;
+ }
+ const totalFieldValues = Object.keys(fieldValues).filter(
+   (fieldName) =>allFields.includes(fieldName)
+ );
+ if (totalFieldValues.length > 0) {
+   return true;
+ }
+
+ return false;
 };
 
 const mapRequiredValidationErrors = (validationErrors, fields) => {
-  const mappedErrors = validationErrors;
-  const allRequiredFields = requiredFieldsArray(fields);
+ const mappedErrors = validationErrors;
+ const allRequiredFields = requiredFieldsArray(fields);
 
-  mappedErrors.errorList = filterErrorList(validationErrors.errorList, allRequiredFields);
+ mappedErrors.errorList = filterErrorList(validationErrors.errorList,
+allRequiredFields);
 
-  return {
-    ...generateErrorSummary(
-      mappedErrors,
-      errorHref,
-    ),
-  };
+ return {
+   ...generateErrorSummary(mappedErrors, errorHref),
+ };
 };
 
 /*
-  ALWAYS_SHOW_ERROR_FIELDS are fields that are:
-  - not required (therefore optional)
-  - BUT if this field has an error, always show it after form submit.
-  - this is the opposite of 'page specific validation rules'
-  - this is currently only used for Companies House validation.
+ ALWAYS_SHOW_ERROR_FIELDS are fields that are:
+ - not required (therefore optional)
```

Comment:
... Have you got indent spaces set to 1 by chance :p ?

Reply:
Done